### DISCUSSION TITLE
Struggling to understand how to call SymFromAddr

### DISCUSSION THREAD

**Discussion by scottbilas** [dahall/Vanara#324] [Q&A] [answered]
I'm having the hardest time understanding how to give `DbgHelp.SymFromAddr` the `SYMBOL_INFO` it requires, such that I get back the symbol name.

I see some custom marshaling in `SafeAnysizeStringMarshaler<SYMBOL_INFO>` but I don't know the right way to engage it.

My current workaround:

* Copy-paste `SYMBOL_INFO` to my own struct where I adjust `SizeConst` on `Name` to 500
* `si.MaxNameLen = 500`
* `si.SizeOfStruct = (uint)Marshal.SizeOf(si) - si.MaxNameLen*2`.

This gets me back a true result and a valid symbol with the name set properly.

What am I doing wrong with the Vanara `SYMBOL_INFO`? Best I could get out of it was a single character set on `Name`, and I tried many combinations on `MaxNameLen` and `SizeOfStruct`.

**Comment by dahall** [comment, marked as answer]
You're not doing anything wrong. There was a problem with the definition of `SYMBOL_INFO` that I fixed in my latest commit. It will be on NuGet by the end of the week and available in a few minutes on AppVeyor. To simplify use, use the following:
```cs
SYMBOL_INFO si = SYMBOL_INFO.Default; // this sets SizeOfStruct and MaxNameLen appropriately
SymFromAddr(hProc, addr, ref disp, ref si);
```

**Reply by scottbilas** [reply]
Wow, thank you for the fast response. I'll look forward to the updated nuget. Thanks again!
